sycl/test-e2e/FreeFunctionKernels/free_function_kernels_enqueue.cpp

```diff
@@ -0,0 +1,187 @@
+// REQUIRES: aspect-usm_shared_allocations
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// XFAIL: target-native_cpu
+// XFAIL-TRACKER: https://github.com/intel/llvm/issues/20142
+
+// This test checks that free function kernels can be submitted using the
+// enqueued functions defined in the free function kernel extension, namely the
+// single_task and the nd_launch functions that take a queue/handler as an
+// argument. These were added in https://github.com/intel/llvm/pull/19995.
+
+#include <cassert>
+#include <sycl/ext/oneapi/experimental/enqueue_functions.hpp>
+#include <sycl/ext/oneapi/experimental/free_function_traits.hpp>
+#include <sycl/ext/oneapi/free_function_queries.hpp>
+#include <sycl/ext/oneapi/work_group_static.hpp>
+#include <sycl/usm.hpp>
+
+namespace syclext = sycl::ext::oneapi;
+namespace syclexp = sycl::ext::oneapi::experimental;
+
+using accType =
+    sycl::accessor<int, 1, sycl::access_mode::read_write, sycl::target::device,
+                   sycl::access::placeholder::true_t>;
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::single_task_kernel))
+void empty() {}
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void initialize(int *ptr) {
+  size_t Lid = syclext::this_work_item::get_nd_item<1>().get_local_linear_id();
+  ptr[Lid] = Lid;
+}
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::single_task_kernel))
+void successor(int *src, int *dst) { *dst = *src + 1; }
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void square(int *src, int *dst) {
+  size_t Lid = syclext::this_work_item::get_nd_item<1>().get_local_linear_id();
+  dst[Lid] = src[Lid] * src[Lid];
+}
+
+template <typename T>
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void squareWithScratchMemoryTemplated(T *src, T *dst) {
+  size_t Lid = syclext::this_work_item::get_nd_item<1>().get_local_linear_id();
+  T *LocalMem = reinterpret_cast<T *>(syclexp::get_work_group_scratch_memory());
+  LocalMem[Lid] = src[Lid] * src[Lid];
+  dst[Lid] = LocalMem[Lid];
+}
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void squareWithAccessor(accType src, accType dst) {
+  size_t Lid = syclext::this_work_item::get_nd_item<1>().get_local_linear_id();
+  dst[Lid] = src[Lid] * src[Lid];
+}
+
+constexpr int SIZE = 16;
+
+int main() {
+  sycl::queue Q;
+  int *Src = sycl::malloc_shared<int>(SIZE, Q);
+  int *Dst = sycl::malloc_shared<int>(SIZE, Q);
+
+  syclexp::single_task(Q, syclexp::kernel_function_s<empty>{});
```

Comment:
Are you using `syclexp::kernel_function_s<empty>{}` instead of `syclexp::kernel_function<empty>` intentionally?